### PR TITLE
fix: History, delete, search fixes

### PR DESCRIPTION
**Pull request by janosSarusiKis**
# Description

Please provide a meaningful description of what this change will do, or is for.
Bonus points for including links to related issues, other PRs, or technical
references.

Note that by _not_ including a description, you are asking reviewers to do extra
work to understand the context of this change, which may lead to your PR taking
much longer to review, or result in it not being reviewed at all.

## Type of Change

- [x] Bugfix
- [ ] New Feature
- [ ] Breaking Change
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please describe)

## Checklist

- [ ] I have read the [contributing guidelines](/agntcy/workflow-srv/blob/main/docs/CONTRIBUTING.md)
- [ ] Existing issues have been referenced (where applicable)
- [ ] I have verified this change is not present in other open pull requests
- [ ] Functionality is documented
- [ ] All code style checks pass
- [ ] New code contribution is covered by automated tests
- [ ] All new and existing tests pass

### CODE REVIEW

File: src/agent_workflow_server/apis/threads.py [hunk 1]
```diff
@@ -223,4 +223,15 @@ async def search_threads(
     thread_search_request: ThreadSearchRequest = Body(None, description=""),
 ) -> List[Thread]:
     """Search for threads.  This endpoint also functions as the endpoint to list all threads."""
-    return await Threads.search(thread_search_request)
+    ## Create filtes from metadata, values and status but only if they are not None
```
Comment (mtrinell) [inline src/agent_workflow_server/apis/threads.py:226]:
```suggestion
    # Create filtes from metadata, values and status but only if they are not None
```

File: tests/test_threads.py [hunk 2]
```diff
@@ -174,12 +174,12 @@ async def test_update_thread(mock_thread, mock_agent):
 @pytest.mark.asyncio
 async def test_search(mock_thread):
     # Test searching for threads
-    threads = await Threads.search({"status": "idle"})
+    threads = await Threads.search(filters ={"status": "idle"}, limit=10, offset=0)
```
Comment (mtrinell) [inline tests/test_threads.py:177]: Can you also include tests without limit/offset (and other filters) ? (i.e. = None)

**Review by mtrinell — APPROVED** [review]
lgtm. Would be nice to add some more tests with None filters

**Review by jubarbot-cisco — APPROVED** [review]
LGTM !

**Review by jubarbot-cisco — APPROVED** [review]
LGTM !
